src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -438,6 +438,17 @@ case class IndexLogEntry(
     properties: Map[String, String])
     extends LogEntry(IndexLogEntry.VERSION) {
 
+  /*
+   * Include the project version in the index log entry file.
+   * This may help prevent in mismatching of project version and index
+   * log entry files.
+   * TODO: breaking change remove if condition. Since currently there are index
+   *  files that don't have project version info and we need to support them as well.
+   */
+  if (properties.contains(IndexLogEntry.HYPERSPACE_PROJECT_VERSION)) {
+    require(properties(IndexLogEntry.HYPERSPACE_PROJECT_VERSION) == BuildInfo.version)
```

Comment:
hmm isn't this too restrictive as I said in the other comment? If there is no breaking change from version 0.5 to 0.6, this check will fail unnecessarily when 0.6 is reading 0.5 log entry.